source/simulator/assembly.cc

```diff
@@ -1551,6 +1636,906 @@ namespace aspect
+                        for (unsigned int j=0; j<advection_dofs_per_cell; ++j)
+                          {
+                            data.local_matrix(i,j)
+                            += (- time_step *  conductivity
+                                * scratch.face_grad_phi_field[i]
+                                * scratch.face_finite_element_values.normal_vector(q)
+                                * scratch.face_phi_field[j]
+
+                                - time_step *  conductivity
+                                * scratch.face_grad_phi_field[j]
+                                * scratch.face_finite_element_values.normal_vector(q)
+                                * scratch.face_phi_field[i]
+
+                                + time_step
+                                * (density_c_P + latent_heat_LHS)
+                                * penalty
+                                * scratch.face_phi_field[i]
+                                * scratch.face_phi_field[j]
+
+                                + (inflow
+                                   ?
+                                   - (density_c_P + latent_heat_LHS)
+                                   * time_step
+                                   * (current_u
+                                      * scratch.face_finite_element_values.normal_vector(q))
+                                   * scratch.face_phi_field[i]
+                                   * scratch.face_phi_field[j]
+                                   :
+                                   0.)
+
+                               )
+                               * scratch.face_finite_element_values.JxW(q);
+
+                          }
+                      }
+                  }
+              }
+            else
+              {
+                //Neumann temperature term - no non-zero contribution as only homogeneous Neumann boundary conditions are implemented elsewhere for temperature
+              }
+          }
+        else
+          {
+            //interior face - no contribution on RHS
+            Assert (cell->neighbor(f).state() == IteratorState::valid,
+                    ExcInternalError());
+            const typename DoFHandler<dim>::cell_iterator neighbor = cell->neighbor(f); //note: NOT active_cell_iterator, so this includes cells that are refined.
+
+            if (face->has_children()) //work from this side, looping over the subfaces
+              {
+                //how does the neighbor talk about this cell?
+                const unsigned int neighbor2 = cell->neighbor_face_no(f);
+
+                //loop over subfaces
+                for (unsigned int subface_no=0; subface_no<face->number_of_children(); ++subface_no)
+                  {
+                    const typename DoFHandler<dim>::cell_iterator neighbor_child      //note: NOT active_cell_iterator, so this includes cells that are refined.
+                      = cell->neighbor_child_on_subface (f, subface_no);
```

Comment:
You should be able to make this an active cell iterator -- the children of an active neighbor must be active.